src/main/java/org/apache/commons/lang3/reflect/TypeUtils.java

```diff
@@ -1018,14 +1028,14 @@ private static Type substituteTypeVariables(final Type type, final Map<TypeVaria
 
         // of the cls's type variables that are arguments of parameterizedType,
         // find out which ones can be determined from the super type's arguments
-        final TypeVariable<?>[] typeVars = getRawType(parameterizedType).getTypeParameters();
+        final TypeVariable<?> @SameLen("typeArgs") [] typeVars = getRawType(parameterizedType).getTypeParameters(); // #5
 
         // use List view of type parameters of cls so the contains() method can be used:
         final List<TypeVariable<Class<T>>> typeVarList = Arrays.asList(cls
                 .getTypeParameters());
 
-        for (int i = 0; i < typeArgs.length; i++) {
-            final TypeVariable<?> typeVar = typeVars[i];
+        for (@IndexOrHigh({"typeVars","typeArgs"}) int i = 0; i < typeArgs.length; i++) { // #6
```

Comment:
This code should just work without any modification given the `@SameLen` type above.